notebooks/ch12/linear_case_study.md

```diff
@@ -0,0 +1,1003 @@
+
+
+```python
+donkeys.columns.values
+```
+
+
+
+
+    array(['BCS', 'Age', 'Sex', 'Length', 'Girth', 'Height', 'Weight',
+           'WeightAlt'], dtype=object)
+
+
+
+Our analysis can be guided by a good understanding of our data, so we should aim to understand what each of these columns represent. A few of these columns are self-explanatory, but others require a little more explanation:
+
+- `BCS`: Body Condition Score 
+- `Girth`: the measurement around the middle of the donkey
+- `WeightAlt`: 31 donkeys within our data frame were weighed twice in order to check the accuracy of the scale. The second weighing is in `WeightAlt`.
+
+
+## Data Cleaning
+
+In this section, we will check the data for any abnormalities that we have to deal with.
+
+By examining `WeightAlt` more closely, we can make sure that the scale is accurate by taking the difference between the two different weighings and plotting them.
+
+
+```python
+difference = donkeys['WeightAlt'] - donkeys['Weight']
+sns.distplot(difference.dropna());
+```
+
+
+![png](linear_case_study_files/linear_case_study_11_0.png)
+
+
+The measurements are all within 1 kg of each other, which seems reasonable.
+
+Next, we can look for unusual values that might indicate errors or other problems. We can use the quantile function in order to detect anomalous values.
+
+
+```python
+donkeys.quantile([0.005, 0.995])
+```
+
+
+
+
+<div>
+<style>
+    .dataframe thead tr:only-child th {
+        text-align: right;
+    }
+
+    .dataframe thead th {
+        text-align: left;
+    }
+
+    .dataframe tbody tr th {
+        vertical-align: top;
+    }
+</style>
+<table border="1" class="dataframe">
+  <thead>
+    <tr style="text-align: right;">
+      <th></th>
+      <th>BCS</th>
+      <th>Length</th>
+      <th>Girth</th>
+      <th>Height</th>
+      <th>Weight</th>
+      <th>WeightAlt</th>
+    </tr>
+  </thead>
+  <tbody>
+    <tr>
+      <th>0.005</th>
+      <td>1.5</td>
+      <td>71.145</td>
+      <td>90.000</td>
+      <td>89.0</td>
+      <td>71.715</td>
+      <td>98.75</td>
+    </tr>
+    <tr>
+      <th>0.995</th>
+      <td>4.0</td>
+      <td>111.000</td>
+      <td>131.285</td>
+      <td>112.0</td>
+      <td>214.000</td>
+      <td>192.80</td>
+    </tr>
+  </tbody>
+</table>
+</div>
+
+
+
+For each of these numerical columns, we can look at which rows fall outside of these quantiles and what values they take on. Consider that we want our model to apply to only healthy and mature donkeys.
+
+
+```python
+donkeys[(donkeys['BCS'] < 1.5) | (donkeys['BCS'] > 4)]['BCS']
+```
+
+
+
+
+    291    4.5
+    445    1.0
+    Name: BCS, dtype: float64
+
+
+
+Considering that `BCS` is an indication of the health of a donkey, a `BCS` of 1 represents an extremely emaciated donkey and a `BCS` of 4.5 an overweight donkey. Thus, it would probably be a good idea to remove these two donkeys.
```

Comment:
Can you also plot the distribution of BCS?

Reply:
mmm considering that we don't plot histograms from the others, i don't think it's necessary :o feel free to fight me on this though

Reply:
I like Tiffany's suggestion. Maybe a table of counts would be good here. Then it would be obvious that there is only 1 donkey with a BCS of 1 and 1 with a BCS of 4.5.